tests/snappi/test_vlan_traffic.py

```diff
@@ -0,0 +1,169 @@
+            "op": "set",
+            "type": "SAI_OBJECT_TYPE_PORT",
+            "attributes": [
+                "SAI_PORT_ATTR_ADMIN_STATE",
+                "true",
+                "SAI_PORT_ATTR_PORT_VLAN_ID",
+                "10",
+            ],
+        },
+        {
+            "name": "bridge_p_3",
+            "op": "create",
+            "type": "SAI_OBJECT_TYPE_BRIDGE_PORT",
+            "attributes": [
+                "SAI_BRIDGE_PORT_ATTR_TYPE",
+                "SAI_BRIDGE_PORT_TYPE_PORT",
+                "SAI_BRIDGE_PORT_ATTR_PORT_ID",
+                "$PORT_3",
+                "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE",
+                "true",
+                "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE",
+                "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
+            ],
+        },
+        {
+            "name": "vlan_member_3",
+            "op": "create",
+            "type": "SAI_OBJECT_TYPE_VLAN_MEMBER",
+            "attributes": [
+                "SAI_VLAN_MEMBER_ATTR_VLAN_ID",
+                "$vlan_10",
+                "SAI_VLAN_MEMBER_ATTR_BRIDGE_PORT_ID",
+                "$bridge_p_3",
+                "SAI_VLAN_MEMBER_ATTR_VLAN_TAGGING_MODE",
+                "SAI_VLAN_TAGGING_MODE_UNTAGGED",
+            ],
+        },
+        {
+            "name": "PORT_3",
+            "op": "set",
+            "type": "SAI_OBJECT_TYPE_PORT",
+            "attributes": [
+                "SAI_PORT_ATTR_ADMIN_STATE",
+                "true",
+                "SAI_PORT_ATTR_PORT_VLAN_ID",
+                "10",
+            ],
+        },
+    ]
+    for command in commands:
+        print("-" * 80)
+        print(command)
+        result = npu.command_processor.process_command(command)
+        print(result)
+
+    macs = ["00:11:11:11:11:11", "00:22:22:22:22:22"]
+    config = dataplane.configuration
+    config.options.port_options.location_preemption = True
+    layer1 = config.layer1.layer1()[-1]
+    layer1.name = "port settings"
+    layer1.port_names = [port.name for port in config.ports]
+    layer1.ieee_media_defaults = False
+    layer1.auto_negotiation.rs_fec = False
+    layer1.auto_negotiation.link_training = False
+    layer1.speed = "speed_100_gbps"
+    layer1.auto_negotiate = False
+    flow1 = config.flows.flow(name="Vlan Traffic")[-1]
+
+    flow1.tx_rx.port.tx_name = config.ports[0].name
+    flow1.tx_rx.port.rx_name = config.ports[1].name
+    flow1.size.fixed = 1024
+    flow1.rate.percentage = 100
+    flow1.metrics.enable = True
+    flow1.metrics.loss = True
+    source = macs[1]
+    destination = macs[0]
+    eth, vlan = flow1.packet.ethernet().vlan()
+    eth.src.value = source
+    eth.dst.value = destination
+    vlan.id.value = int(vlan_id)
+
+    dataplane.set_config()
+    restpy_session = dataplane.api.assistant.Session
+    ixnet = restpy_session.Ixnetwork
+    for port in ixnet.Vport.find():
+        port.L1Config.NovusHundredGigLan.AutoInstrumentation = "endOfFrame"
```

Comment:
This looks like a vendor-specific attribute. Can we avoid using them? Can we write the tests so that they can also run with ixia-c?